website/static/js/setPassword.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+
+var ko = require('knockout');
+require('knockout.validation');
+var $ = require('jquery');
+var zxcvbn = require('zxcvbn');
+
+var $osf = require('./osfHelpers');
+
+
+// Accepts a few different types of arguments, depending on
+// the desired fields and type of password reset.
+// types:
+//    - signup: required fields for Name, email, email confirmation, password, and password strength
+//    - reset: fields for password, strength, and password confirmation (for when you don't know your old password)
+//    - change: change your password when you know your old one
+var ViewModel = function(passwordViewType, submitUrl, campaign) {
+
+    var self = this;
+
+    self.typedPassword = ko.observable('');
+
+    self.passwordInfo = ko.computed(function() {
+        if (self.typedPassword()) {
+            return zxcvbn(self.typedPassword());
```

Comment:
While testing very long passwords, somewhere around 180 characters my browser stopped responding. the [zxcvbn performance page](https://github.com/dropbox/zxcvbn#performance) suggests sending no more than 100 characters to zxcvbn to minimize runtime latency.